pkg/controller/combo_controller.go

```diff
@@ -2,36 +2,161 @@ package controller
+			c.log.Info(fmt.Sprintf("enqueueing %s combination in response to associated %s template being updated", combination.Name, templateName))
+			requests = append(requests, reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      combination.Name,
+					Namespace: template.GetNamespace(),
+				},
+			})
+		}
+	}
+
+	return requests
+}
+
+// Reconcile manages incoming combination CR's and processes them accordingly
 func (c *combinationController) Reconcile(ctx context.Context, req ctrl.Request) (reconcile.Result, error) {
-	// Set up a convenient log object so we don't have to type request over and over again
+	// Set up a convenient log object so we don’t have to type request over and over again
 	log := c.log.WithValues("request", req)
-	log.Info("reconciling combination")
 
-	in := &v1alpha1.Combination{}
-	if err := c.Get(ctx, req.NamespacedName, in); err != nil {
+	log.V(1).Info("new combination event inbound")
+
+	// Attempt to retrieve the requested combination CR
+	combination := &v1alpha1.Combination{}
+	err := c.Get(ctx, req.NamespacedName, combination)
+	if err != nil {
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
-	return reconcile.Result{}, nil
+	// If combination is being deleted, remove from queue
+	if !combination.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.Info("combination is being deleted, ignoring event")
+		return reconcile.Result{}, nil
+	}
+
+	// Attempt to retrieve the template referenced in the combination CR
+	template := &v1alpha1.Template{}
+	if err := c.Get(ctx, types.NamespacedName{Name: combination.Spec.Template}, template); err != nil {
+		combination.SetCondition(metav1.Condition{
+			Type:               v1alpha1.TypeInvalid,
+			Status:             metav1.ConditionFalse,
+			Reason:             v1alpha1.ReasonTemplateNotFound,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+			Message:            fmt.Sprintf("failed to retrieve %s template: %s", combination.Spec.Template, err.Error()),
+		})
+		return reconcile.Result{}, errors.NewAggregate([]error{err, c.Status().Update(ctx, combination)})
+	}
+
+	// Build combination stream to be utilized in template builder
+	comboStream := combinationPkg.NewStream(
+		combinationPkg.WithArgs(formatArguments(combination.Spec.Arguments)),
+		combinationPkg.WithSolveAhead(),
+	)
+
+	// Create a new template builder
+	builder, err := templatePkg.NewBuilder(strings.NewReader(template.Spec.Body), comboStream)
+	if err != nil {
+		combination.SetCondition(metav1.Condition{
+			Type:               v1alpha1.TypeInvalid,
+			Status:             metav1.ConditionFalse,
+			Reason:             v1alpha1.ReasonTemplateBodyInvalid,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+			Message:            fmt.Sprintf("failed to construct a builder out of %s template body:  %s", combination.Spec.Template, err.Error()),
+		})
+		return reconcile.Result{}, errors.NewAggregate([]error{err, c.Status().Update(ctx, combination)})
+	}
```

Comment:
I appreciate that you "log" via the status instead of printing output. Nice work!